### PR TITLE
Add response.statusMessage data information in environment logs

### LINKED ISSUE #852: Add `response.statusMessage` data information in environment logs
### Description:
    As a user of desktop application or CLI version
I want to see status message [status message of response](https://nodejs.org/api/http.html#responsestatusmessage).
Curl verbose output with example, what I'm talking about:
image

### Cause:
Currently I'm using mockoon in Proxy mode for capturing any request to some external service.
That service is using status message as a place of sending information about any error, caused by user, or by server (I mean resposes with 400+ status code). In logs there are no such important information for me, that makes me to retry request manually for getting this status, so my efforts for testing is doubled in that case.

### PR DESCRIPTION
**Pull request by AndreiOrmanji**
Closes #852

**Technical implementation details**
I've been added a new property  `statusMessage` in `Transaction.response` located in `packages\commons\src\models\server.model.ts` for getting the statusMessage in "Logs" tab and some other places:
 - `function CreateTransaction` in packages\commons-server\src\libs\utils.ts that converts a pair of `Request` and `Response` objects in a `Transaction` object
 - `EnvironmentLogResponse` in packages\desktop\src\renderer\app\models\environment-logs.model.ts for showing the information from modified `Transaction` in "Logs" tab
 - `function formatLog` in packages\desktop\src\renderer\app\services\data.service.ts for converting the `Transaction` object in 
`EnvironmentLogResponse` object.
- added a new item in "General" section of "Response" tab in packages\desktop\src\renderer\app\components\environment-logs\environment-logs.component.html
- updated existing tests for covering my changes.

**Disclaimer:**
I'm not an experienced open software contributor or at least a JS developer, but I want to help you, Mockoon supporters, to make Mockoon better.

**Checklist**

- [ ] data migration added (@mockoon/commons)
- [ ] data migration automated tests added (@mockoon/commons)
- [ ] CLI automated tests added (@mockoon/cli)
- [ ] desktop automated tests added (@mockoon/desktop)

### CODE REVIEW

**Comment by AndreiOrmanji** [comment]
I don't know if I'm allowed to mark any item in checklist as "solved", so I've left it all "as is".

File: packages/desktop/src/renderer/app/components/environment-logs/environment-logs.component.html [hunk 1]
```diff
@@ -328,6 +328,9 @@
                   <div class="environment-logs-content-item">
                     <strong>Status:</strong> {{ selectedLog.response?.status }}
                   </div>
+                  <div class="environment-logs-content-item">
+                    <strong>StatusMessage:</strong> {{ selectedLog.response?.statusMessage }}
```
Comment (255kb) [inline packages/desktop/src/renderer/app/components/environment-logs/environment-logs.component.html:332]: Maybe title could be more human readable:
```suggestion
                    <strong>Status message:</strong> {{ selectedLog.response?.statusMessage }}
```

**Review by 255kb — CHANGES_REQUESTED** [review]
Thanks for the contribution. 
I run the CI and there are some Prettier formatting issue. Please run Prettier before pushing again. I also add a comment concerning the log label.

**Comment by 255kb** [comment]
Now that I am thinking about it, I think it would be better to display the Status message next to the status code, so it's more in line with what other tools like Curl are displaying. What do you think?

**Comment by AndreiOrmanji** [comment]
> Now that I am thinking about it, I think it would be better to display the Status message next to the status code, so it's more in line with what other tools like Curl are displaying. What do you think?

In my optinion,it will "break existing UI consistency" where you have already 'General', 'Headers', 'Body' sections.
If you are ok with adding more details, I can add HTTP version, send in response by server (probably can be useful for other users).

**Comment by 255kb** [comment]
> > Now that I am thinking about it, I think it would be better to display the Status message next to the status code, so it's more in line with what other tools like Curl are displaying. What do you think?
> 
> In my optinion,it will "break existing UI consistency" where you have already 'General', 'Headers', 'Body' sections. If you are ok with adding more details, I can add HTTP version, sent in response by server (probably can be useful for other users).

What I meant was to add the message next to the status code to avoid adding a new line, like this:
```
Status: 404 - Not Found
```

Instead of:
```
Status: 404
Status message: Not Found
```
I'm not sure if adding the HTTP version is really useful. Wouldn't it be always something like 1.1 as we are using Node.js' http?

**Comment by AndreiOrmanji** [comment]
> > > Now that I am thinking about it, I think it would be better to display the Status message next to the status code, so it's more in line with what other tools like Curl are displaying. What do you think?
> > 
> > 
> > In my optinion, it will "break existing UI consistency" where you have already 'General', 'Headers', 'Body' sections. If you are ok with adding more details, I can add HTTP version, sent in response by server (probably can be useful for other users).
> 
> What I meant was to add the message next to the status code to avoid adding a new line, like this:
> 
> ```
> Status: 404 - Not Found
> ```
> 
> Instead of:
> 
> ```
> Status: 404
> Status message: Not Found
> ```

Your varians looks good too :)


> I'm not sure if adding the HTTP version is really useful. Wouldn't it be always something like 1.1 as we are using Node.js' http?

It depends on server the requests are sent to. For example, persistent connection (read more at https://developer.mozilla.org/en-US/docs/Web/HTTP/Connection_management_in_HTTP_1.x) is available from 1.1 and higher, so it can be useful to know what version is supported by server.

**Comment by AndreiOrmanji** [comment]
> If you don't mind, maybe display status code and message on one line, I think I prefer 🙂 Also, if you want to display the HTTP version go for it!

I want to add HTTP version. Can you add an example how it would look like?

**Comment by 255kb** [comment]
Maybe something like:

```
Protocol: HTTP1.1
Status: 404 - Not found
```

So it's ready for other protocols like web socket **when** we will implement web sockets 😄.

**Comment by AndreiOrmanji** [comment]
> Maybe something like:
> 
> ```
> Protocol: HTTP1.1
> Status: 404 - Not found
> ```
> 
> So it's ready for other protocols like web socket **when** we will implement web sockets smile.

I've implemented like so:
image.

**Comment by AndreiOrmanji** [comment]
Sorry for pipeline fails, but I could not set up project on my machine to execute tests from pipeline...
Fixed problem caused pipeline fail.

**Comment by 255kb** [comment]
No worries, I used the CI all the time to run the tests 😅
There are still many test that fail, apparently for an extra ":". 
You can run the test locally by running first `npm run build:desktop:ci` then `npm run test:desktop -- --spec "environment-logs.spec.ts"`.
Let me know if you need help with anything.

**Comment by AndreiOrmanji** [comment]
It seems that node js and express did not provide access to protocol and version to send in response, so I've set values from request, where it was possible.
I would remove `Protocol` section because i'm not sure it was implemented in a "correct" way and to be implemented by someone  who has more experience with JS and node.

I've refused to use docker. With windows I've the following error on test running:
```bash
$ npm -v
8.19.2

$ node -v
v19.0.0

$ nvm -v
Running version 1.1.9.
```
Screenshot:
image

**Comment by 255kb** [comment]
> I would remove `Protocol` section because i'm not sure it was implemented in a "correct" way and to be implemented by someone who has more experience with JS and node.

I would say the same. Maybe remove it and stick to the status message only? It's probably too much hassle for the benefits right now.

**Comment by AndreiOrmanji** [comment]
> Maybe remove it and stick to the status message only? It's probably too much hassle for the benefits right now.

Removed changes, related to `Protocol`, so the current version looks like:
image

**Review by 255kb — APPROVED** [review]

**Comment by 255kb** [comment]
Thank you for the changes. Everything looks ok, I will merge and add it to the next release (probably later this month).

**Comment by AndreiOrmanji** [comment]
Thank you very much

**Pull request merged**